test/jdk/jdk/incubator/vector/AbstractVectorConversionTest.java

```diff
@@ -245,22 +247,20 @@ else if (src == double.class) {
         List<Object[]> args = new ArrayList<>();
         for (Class<?> srcE : List.of(byte.class, short.class, int.class, long.class, float.class, double.class)) {
             VectorSpecies<?> src = VectorSpecies.of(srcE, srcShape);
-            List<VectorSpecies> sps = new ArrayList<VectorSpecies>();
             for (VectorShape dstShape : VectorShape.values()) {
                 for (Class<?> dstE : List.of(byte.class, short.class, int.class, long.class, float.class, double.class)) {
                     VectorSpecies<?> dst = VectorSpecies.of(dstE, dstShape);
                     if (legal) {
```

Comment:
I think we just need to do `xnor` on `legal` and `dst.length() == src.length()`, we could simplify with:
```
// add if legal and equal lengths, or illegal and unequal lengths 
if (legal == (dst.length() == src.length())) {
    args.add(new Object[]{src, dst});
}
```